web3.js/src/vote-account.ts

```diff
@@ -121,15 +160,49 @@ export class VoteAccount {
 
     return new VoteAccount({
       nodePubkey: new PublicKey(va.nodePubkey),
-      authorizedVoterPubkey: new PublicKey(va.authorizedVoterPubkey),
-      authorizedWithdrawerPubkey: new PublicKey(va.authorizedWithdrawerPubkey),
+      authorizedWithdrawer: new PublicKey(va.authorizedWithdrawer),
       commission: va.commission,
       votes: va.votes,
       rootSlot,
-      epoch: va.epoch,
-      credits: va.credits,
-      lastEpochCredits: va.lastEpochCredits,
+      authorizedVoters: va.authorizedVoters.map(parseAuthorizedVoter),
+      priorVoters: getPriorVoters(va.priorVoters),
       epochCredits: va.epochCredits,
+      lastTimestamp: va.lastTimestamp,
     });
   }
 }
+
+function parseAuthorizedVoter({epoch, authorizedVoter}: AuthorizedVoter) {
+  return {
+    epoch,
+    authorizedVoter: new PublicKey(authorizedVoter),
+  };
+}
+
+function parsePriorVoters({
+  authorizedPubkey,
+  epochOfLastAuthorizedSwitch,
+  targetEpoch,
+}: PriorVoter) {
+  return {
+    authorizedPubkey: new PublicKey(authorizedPubkey),
+    epochOfLastAuthorizedSwitch,
+    targetEpoch,
+  };
+}
+
+function getPriorVoters({
+  buf,
+  idx,
+  isEmpty,
+}: {
+  buf: PriorVoter[];
+  idx: number;
+  isEmpty: boolean;
+}): PriorVoter[] {
+  if (isEmpty) {
+    return [];
+  }
+
+  return [...buf.slice(idx).map(parsePriorVoters), ...buf.slice(0, idx - 1)];
```

Comment:
Sorry, I misled you. This should be:
```suggestion
  return [...buf.slice(idx + 1).map(parsePriorVoters), ...buf.slice(0, idx)];
```

Reply:
This will ensure that the last "prior voter" is always the current one

Reply:
Yup, this makes sense